watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -330,14 +362,41 @@ def build(self):
             units=pyunits.m,
             doc="Stokes radius of solute",
         )
-        self.diffus_phase_comp = Param(
-            self.phase_list,
-            self.ion_set | self.solute_set,
+        if self.config.diffus_calculation == DiffusivityCalculation.none:
+            self.diffus_phase_comp = Param(
+                self.phase_list,
+                self.ion_set | self.solute_set,
+                mutable=True,
+                default=1e-9,
+                initialize=self.config.diffusivity_data,
+                units=pyunits.m**2 * pyunits.s**-1,
+                doc="Bulk diffusivity of solute components",
+            )
+        elif self.config.diffus_calculation == DiffusivityCalculation.HaydukLaudie:
+            hl_ind = [i[1] for i in self.config.molar_volume_data.keys()]
+            nonhl_ind = [i[1] for i in self.config.diffusivity_data.keys()]
```

Comment:
Add comments above each to spell out what you're doing (assuming hl_ind gives you components in solute_set and nonhl_ind give ion_set).

Reply:
I would also make the check more explicit, e.g., check `hl_ind` against `solute_set` and `nonhl_ind` against `ion_set`. Not sure intermediate lists are needed either.

What is stored in `i[0]`?

Reply:
We were discussing around a situation where we may allow NOT all components in the given `solute_list` to possess a `diffus_phase_comp` property.  I am having about-ready  commits to realize this but have one pending decision to make-  for those which won't have a diffusivity data, do we not build the property at all or have then lingering with default values.  I am in favor of not building them to avoid confusions or potential mistakes.  Maybe we can try deciding this in the dev meeting if a few min of discussion would be enough.

Reply:
`I[0]` is `"Liq"` (of course ... :))

>

Reply:
I would recommend caution on this - whilst it makes sense if you know what is going on, if makes things harder for downstream models as they may not know which components have a diffusivity and then they need to know what to do if one does not exist.